spacelift/resource_scheduled_task.go

```diff
@@ -0,0 +1,270 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+type ScheduledTask struct {
+	At       *int
+	Command  string
+	Every    []string
+	Timezone string
+}
+
+func resourceScheduledTask() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_scheduled_task` represents a scheduling configuration " +
+			"for a Stack. It will trigger task on the given schedule or timestamp",
+
+		CreateContext: resourceScheduledTaskCreate,
+		ReadContext:   resourceScheduledTaskRead,
+		UpdateContext: resourceScheduledTaskUpdate,
+		DeleteContext: resourceScheduledTaskDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"stack_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack for which to set up the scheduled task",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"command": {
+				Type:             schema.TypeString,
+				Description:      "Command that will be run.",
+				Required:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+			},
+			"at": {
+				Type:             schema.TypeInt,
+				Description:      "Timestamp (unix timestamp) at which time the scheduled task should happen.",
+				Optional:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+				ConflictsWith:    []string{"every"},
```

Comment:
I know you've done it in the opposite direction, but should this include `timezone` as well?

Reply:
Sure